Mercury-Store/Mercury-Store/Support/CoreData/FavouriteCoreDataModel.swift

```diff
@@ -17,6 +17,16 @@ extension CoreDataModel{
             return delete(itemID: productID)
         }
     }
+    func toggleFavourite(product:SavedProductItem)->Bool{
+        let state = !isProductFavourite(id: Int(truncating: NSDecimalNumber(decimal: product.productID) ))
+        if (!state) {
```

Comment:
```suggestion
        let isFavorite = isProductFavourite(id: Int(truncating: NSDecimalNumber(decimal: product.productID) ))
        if isFavorite {
```

Reply:
Why this conversion? `Int(truncating: NSDecimalNumber(decimal: product.productID)`